frontend/src/advising/Templates/TemplateInfoPage.tsx

```diff
@@ -67,9 +83,150 @@ interface DropdownProps {
 
 interface NameFieldProps {
   readonly name: string;
-  readonly setTemplateName: (name: string) => void;
+  readonly setName: (name: string) => void;
+  readonly label: string;
+  readonly error: boolean;
 }
 
+interface FolderContext {
+  readonly folders: IFolderData[];
+  readonly setSelectedFolderId: (selectedFolderId: number | null) => void;
+  readonly newFolderName: string;
+  readonly setNewFolderName: (newFolderName: string) => void;
+  readonly hasDuplicateFolderName: boolean;
+}
+
+const folderContext = createContext<Partial<FolderContext>>({});
+
+const NameField: React.FC<NameFieldProps> = ({
+  name,
+  setName,
+  label,
+  error,
+}) => {
+  return (
+    <TextField
+      id="outlined-basic"
+      label={label}
+      variant="outlined"
+      value={name}
+      onChange={event => setName(event.target.value)}
+      placeholder=""
+      style={{ width: "100%" }}
+      error={error}
+    />
+  );
+};
+
+const DisplayedFolderInputField: React.FC<{ creatingNewFolder: boolean }> = ({
+  creatingNewFolder,
+}) => {
+  const {
+    folders,
+    setSelectedFolderId,
+    newFolderName,
+    setNewFolderName,
+    hasDuplicateFolderName,
+  } = useContext(folderContext) as FolderContext;
+
+  useEffect(() => {
+    setSelectedFolderId(null);
+    setNewFolderName("");
+  }, [creatingNewFolder]);
+
+  // TODO: FIx new folder input being a greater height than existing folder input
+  if (creatingNewFolder) {
+    return (
+      <FormControl
+        variant="outlined"
+        error={hasDuplicateFolderName}
+        style={{ width: INPUT_WIDTH }}
+      >
+        <NameField
+          name={newFolderName}
+          setName={setNewFolderName}
+          label={"New folder name"}
+          error={hasDuplicateFolderName}
+        />
+        <FormHelperText>
+          {hasDuplicateFolderName && "Folder name has already been taken"}
+        </FormHelperText>
+      </FormControl>
+    );
+  }
+
+  return (
+    <FormControl variant="outlined">
+      <Autocomplete
+        style={{ width: INPUT_WIDTH }}
+        disableListWrap
+        getOptionLabel={option => option.name}
+        options={folders}
+        renderInput={params => (
+          <TextField
+            {...params}
+            variant="outlined"
+            label={"Existing folder"}
+            fullWidth
+          />
+        )}
+        onChange={(event, newValue: any) =>
+          setSelectedFolderId(newValue.id || null)
+        }
+      />
+    </FormControl>
+  );
+};
+
+const FolderSelection: React.FC<{}> = () => {
```

Comment:
The entire folder selection component including the picker between both options and the input itself